ports/cheriot-rtos/machine_spi.c

```diff
@@ -0,0 +1,300 @@
+#include <stdio.h>
+
+#include "extmod/modmachine.h"
+#include "py/obj.h"
+#include "py/runtime.h"
+
+#define STRINGIFY(x) #x
+#define EX_STRINGIFY(x) STRINGIFY(x)
+
+#define SPI_MAX_BAUDRATE (25000000)
+#define SPI_MIN_BAUDRATE (SPI_MAX_BAUDRATE >> 16)
+#define DEFAULT_SPI_BAUDRATE (1000000)
+#define DEFAULT_SPI_POLARITY (0)
+#define DEFAULT_SPI_PHASE (0)
+#define DEFAULT_SPI_BITS (8)
+#define DEFAULT_SPI_FIRSTBIT (SPI_MSB_FIRST)
+
+typedef struct _spi_block_t {
+    uint32_t interruptState;
+    uint32_t interruptEnable;
+    uint32_t interruptTest;
+    uint32_t configuration;
+    uint32_t control;
+    uint32_t status;
+    uint32_t start;
+    uint32_t receiveFifo;
+    uint32_t transmitFifo;
+} spi_block_t;
+
+/// Configuration Register Fields
+enum {
+    ConfigurationHalfClockPeriodMask = 0xffffu << 0,
+    ConfigurationMSBFirst = 1u << 29,
+    ConfigurationClockPhase = 1u << 30,
+    ConfigurationClockPolarity = 1u << 31,
+};
+
+/// Control Register Fields
+enum {
+    ControlTransmitClear = 1 << 0,
+    ControlReceiveClear = 1 << 1,
+    ControlTransmitEnable = 1 << 2,
+    ControlReceiveEnable = 1 << 3,
+    ControlTransmitWatermarkMask = 0xf << 4,
+    ControlReceiveWatermarkMask = 0xf << 8,
+};
+
+/// Status Register Fields
+enum {
+    StatusTxFifoLevel = 0xffu << 0,
+    StatusRxFifoLevel = 0xffu << 8,
+    StatusTxFifoFull = 1u << 16,
+    StatusRxFifoEmpty = 1u << 17,
+    StatusIdle = 1u << 18,
+};
+
+/// Start Register Fields
+enum {
+    StartByteCountMask = 0x7ffu,
+};
+
+void configure_spi_block(volatile spi_block_t *spi, const bool ClockPolarity,
+    const bool ClockPhase, const bool MsbFirst,
+    const uint16_t HalfClockPeriod) {
```

Comment:
According to the style guide, use underscore_case, not camelCase for all names.

Also why are you declaring by-value function parameters as `const`?